Arm/State.lean

```diff
@@ -1022,3 +1074,32 @@ by_cases h : ix < base
   · simp only [h₂, ↓reduceIte, BitVec.getLsb_extractLsByte]
 
 end Memory
+
+/-! ## Helper lemma for `AxEffects` -/
```

Comment:
BTW, `state_simp_rules` tag for these theorems too.

Reply:
I'm not sure these would make good simp-lemmas: they don't really have an obvious direction that makes for a good simp-normal form

Reply:
In particular, if we had both `∀ n addr, m₁.read_bytes n addr = m₂.read_bytes n addr` and the symmetric version `∀ n addr, m₂.read_bytes n addr = m₁.read_bytes n addr` hypotheses, and include both in the simp-set, then these equalities would loop. I think nowadays simp is smart enough to detect such loops, but it's still considered bad form to have a looping simp-set

Reply:
You're absolutely right, thanks!